elytron/src/main/java/org/wildfly/extension/elytron/ElytronSubsystemTransformers.java

```diff
@@ -153,6 +158,26 @@ private static void from17(ChainedTransformationDescriptionBuilder chainedBuilde
                 .getAttributeBuilder()
                 .addRejectCheck(RejectAttributeChecker.DEFINED, TRUSTED_VIRTUAL_SECURITY_DOMAINS)
                 .setDiscard(DiscardAttributeChecker.UNDEFINED, TRUSTED_VIRTUAL_SECURITY_DOMAINS);
+
+        builder.addChildResource(PathElement.pathElement(MAPPED_ROLE_MAPPER))
+                .getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, FROM)
+                .setDiscard(DiscardAttributeChecker.UNDEFINED, TO)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, FROM)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, TO)
+                .addRejectCheck(new RejectAttributeChecker.DefaultRejectAttributeChecker() {
+                    /* [WFCORE-6244] role-map attribute was previously formatted (directly) as a string to string-list mapping of roles.
+                     * Attributes formatted as a list of maps (ObjectListAttributeDefinition) should be rejected. */
+                    @Override
+                    protected boolean rejectAttribute(PathAddress address, String attributeName, ModelNode attributeValue, TransformationContext context) {
+                        return attributeValue.getType() != ModelType.OBJECT;
+                    }
+
+                    @Override
+                    public String getRejectionLogMessage(Map<String, ModelNode> attributes) {
+                        return ROOT_LOGGER.mappedRoleMapExpectsObject().getMessage();
```

Comment:
The rejection message is not the expected one. 

The attribute is rejected when it is a list of Objects but the Rejection message is:

`Value type for attribute `role-map` is invalid. Expected [LIST] of [OBJECT].`

It should be something that describes what is expected, in this case, something like it is expected a [OBJECT] but got a LIST.

Reply:
@fjuma referring back to [this comment](https://github.com/wildfly/wildfly-core/pull/5393#discussion_r1149454732), I'll restore this message to the format I previously had. 

Should I just try to detect the type of object, and allow the `getType` command to throw an Exception if it fails? Or would it be better to just create an if-else statement, where the alternate message without the type is thrown if 'role-map` attribute doesn't exist?

Reply:
alright, I'll remove the case then, thanks

Reply:
Talked with @fjuma about the issue, i ended up replacing this method with a ValueConverter, since only the model needs to be updated (the schema hasn't changed).

Reply:
ok, that makes more sense and was our original suggestion. I assumed you wanted to reject it instead of doing a conversion.